examples/solid/start-bare/src/routes/__root.tsx

```diff
@@ -1,24 +1,50 @@
-import { createRootRoute, Link, Outlet } from '@tanstack/solid-router'
+import {
+  createRootRoute,
+  HeadContent,
+  Link,
+  Outlet,
+  Scripts,
+} from '@tanstack/solid-router'
 import appCss from '~/styles/app.css?url'
 import * as Solid from 'solid-js'
 import { TanStackRouterDevtools } from '@tanstack/solid-router-devtools'
+import { Hydration, HydrationScript, NoHydration } from 'solid-js/web'
 
 export const Route = createRootRoute({
   head: () => ({
     links: [{ rel: 'stylesheet', href: appCss }],
   }),
-  component: RootDocument,
+  component: RootComponent,
 })
 
-function RootDocument({ children }: { children: Solid.JSX.Element }) {
+function RootComponent() {
   return (
-    <>
-      <div class="p-2 flex gap-2 text-lg">
-        <Link to="/">Index</Link>
-        <Link to="/about">About</Link>
-      </div>
+    <RootDocument>
       <Outlet />
-      <TanStackRouterDevtools position="bottom-right" />
-    </>
+    </RootDocument>
+  )
+}
+
+function RootDocument({ children }: { children: Solid.JSX.Element }) {
+  return (
+    <NoHydration>
+      <html>
+        <head>
+          <HydrationScript />
+          <HeadContent />
+        </head>
+        <body>
+          <Hydration>
```

Comment:
This should probably be similar to SolidStart's approach. The current `<NoHydration>` placement is correct but not for `<Hydration>`. We should pick a pre-rendered root as a hydration root (instead of `document.body`), so something like

```jsx
return (
  <body>
    <div id="app">
      <Hydration>
        {content}
      </Hydration>
    </div>
  </body>
);
```

This should make it easier to pick up on the client. That leaves you the `TanStackRouterDevtools` to handle later (which should be easy, it could be its own hydration root too).